sdk/digitaltwins/azure-digitaltwins/azure/digitaltwins/_digitaltwin_models_client.py

```diff
@@ -0,0 +1,116 @@
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    :**kwargs Used to configure the service client.
+    :type **kwargs: any
+    """
+    def __init__(self, endpoint: str, credential: object, **kwargs: any):
+        # type: (str, AzureKeyCredential, **Any) -> None
+
+        self.endpoint = endpoint #type: str
+        self.credential = credential #type AzureKeyCredential
+        self._client = AzureDigitalTwinsAPI(
+            credential=credential,
+            base_url=endpoint,
+            **kwargs
+        ) #type: AzureDigitalTwinsAPI
+
+    def get_model(self, model_id, include_model_definition=False, **kwargs):
+        # type: (str, bool, **Any) -> "models.ModelData"
+        """Get a model, including the model metadata and the model definition
+
+        :param model_id: The Id of the model
+        :type model_id: str
+        :param include_model_definition: When true the model definition
+        will be returned as part of the result
+        :type include_model_definition: bool
+        :**kwargs The operation options
+        :type **kwargs: any
+        :returns: The ModelDate object
+        :rtype: ~azure.digitaltwins.models.ModelData
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twin_models.get_by_id(model_id, include_model_definition, **kwargs)
+
+    def list_models(self, dependencies_for, include_model_definition=False, max_item_count=-1, **kwargs):
+        # type: (str, bool, int, **Any) -> Iterable["models.PagedModelDataCollection"]
+        """Get the list of models
+
+        :param dependencies_for: The model Ids to have dependencies retrieved.
+        If omitted, all models are retrieved
+        :type dependencies_for: str
+        :param include_model_definition: When true the model definition
+        will be returned as part of the result
+        :type include_model_definition: bool
+        :param max_item_count: The maximum number of items to retrieve per request.
+        The server may choose to return less than the requested max.
+        :type max_item_count: int
+        :**kwargs The operation options
+        :type **kwargs: any
+        :returns: An iterator like instance of either PagedModelDataCollection
+        :rtype: ~azure.core.paging.ItemPaged[~azure.digitaltwins.models.PagedModelDataCollection]
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        digital_twin_models_list_options = {'max_item_count': -1}
+        if max_item_count != -1:
+            digital_twin_models_list_options= {'max_item_count': max_item_count}
+        return self._client.digital_twin_models.list(
+            dependencies_for=dependencies_for,
+            include_model_definition=include_model_definition,
+            digital_twin_models_list_options=digital_twin_models_list_options, **kwargs
+        )
+
+    def create_models(self, models=None, **kwargs):
```

Comment:
The `modesl` parameter is optional? What happens if I call this operation with no input?

Reply:
It will succeed. It is strange but this is how the API is defined.

---
In reply to: [491745408](https://github.com/Azure/azure-sdk-for-python/pull/13852#discussion_r491745408) [](ancestors = 491745408)